File: AndroidStudio/individual10/app/src/main/java/com/example/individual10/MainActivity.java
package com.example.individual10;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private EditText etUrl;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        etUrl = findViewById(R.id.etUrl);
        Button btnOpenWeb = findViewById(R.id.btnOpenWeb);

        btnOpenWeb.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String url = etUrl.getText().toString().trim();

                // Validar que la URL no esté vacía
                if (url.isEmpty()) {
                    Toast.makeText(MainActivity.this, "Por favor, ingrese una URL", Toast.LENGTH_SHORT).show();
                    return;
                }

                // Pasar la URL a la actividad WebViewActivity
                Intent intent = new Intent(MainActivity.this, WebViewActivity.class);
                intent.putExtra("url", url);
                startActivity(intent);
            }
        });
    }
}

File: AndroidStudio/Individual10M5/app/src/main/java/com/example/individual10m5/MainActivity.kt
package com.example.individual10m5

import android.os.Bundle
import android.widget.*
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {


    private val conversionRates = mapOf(
        "USD to EUR" to 0.85,
        "EUR to USD" to 1.18,
        "USD to CLP" to 840.00,
        "CLP to USD" to 0.0012
    )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        val amountEditText = findViewById<EditText>(R.id.amountEditText)
        val currencySpinner = findViewById<Spinner>(R.id.currencySpinner)
        val resultTextView = findViewById<TextView>(R.id.resultTextView)
        val convertButton = findViewById<Button>(R.id.convertButton)
        val resetButton = findViewById<Button>(R.id.resetButton)


        val options = listOf("USD to EUR", "EUR to USD", "USD to CLP", "CLP to USD")
        val adapter = ArrayAdapter(this, android.R.layout.simple_spinner_item, options)
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        currencySpinner.adapter = adapter


        convertButton.setOnClickListener {
            val amount = amountEditText.text.toString().toDoubleOrNull()
            val selectedConversion = currencySpinner.selectedItem.toString()

            if (amount != null && conversionRates.containsKey(selectedConversion)) {
                val conversionRate = conversionRates[selectedConversion] ?: 1.0
                val convertedAmount = amount * conversionRate
                resultTextView.text = "Converted Amount: $convertedAmount"
            } else {
                resultTextView.text = "Invalid input or conversion."
            }
        }


        resetButton.setOnClickListener {
            amountEditText.text.clear()
            resultTextView.text = ""
        }
    }
}